src/Administration/Resources/app/administration/src/module/sw-customer/page/sw-customer-detail/index.js

```diff
@@ -175,6 +186,31 @@ Component.register('sw-customer-detail', {
 
         onActivateCustomerEditMode() {
             this.editMode = true;
-        }
+        },
+
+        checkPassword() {
```

Comment:
The function looks very complicated with these nested `if`-conditions.
And it also does two things. It validates the equality of the input and modifies the `state` of the this-scope with `this.customer.password = this.customer.passwordNew;`


Can you try to refactor this function?  Maybe something in the way like this?

```
            ...
            if (!this.validPassword(this.customer)) {
                 this.isLoading = false;

                 return false;
            } else {
                this.customer.password = this.customer.passwordNew;
            }

        ....

        validPassword(customer) {
            const { passwordNew, passwordConfirm } = customer;
            const bothValuesSet = passwordNew || passwordConfirm;
            const passwordEqualsConfirmation = passwordNew !== passwordConfirm;
            const invalidLength = (passwordNew && passwordNew.length < 8);

            if (bothValuesSet) {
                if (passwordEqualsConfirmation) {
                    this.createNotificationError({
                        title: this.$tc('sw-customer.detail.titleSaveError'),
                        message: this.$tc('sw-customer.detail.notificationPasswordErrorMessage')
                    });

                    return false;
                } else if (invalidLength) {
                    this.createNotificationError({
                        title: this.$tc('sw-customer.detail.titleSaveError'),
                        message: this.$tc('sw-customer.detail.notificationPasswordLengthErrorMessage')
                    });

                    return false;
                }
            }

            return true;
         },
```

Reply:
@raknison I've given it a refactor with your suggestions in mind and included the config for minPasswordLength too.

```
            if (! await this.validPassword(this.customer)) {
                this.isLoading = false;
                return false;
            } else if (this.customer.passwordNew) {
                this.customer.password = this.customer.passwordNew;
            }
```

Here we'll need to check if passwordNew is set after validPassword returns true (which it also does if the password isnt changed). Otherwise it would always change the password on an edit, so you could end up with an empty password :-)


```
        async validPassword(customer) {
            const config = await this.systemConfigApiService.getValues('core.register');

            const { passwordNew, passwordConfirm } = customer;
            const passwordSet = (passwordNew || passwordConfirm);
            const passwordNotEquals = (passwordNew !== passwordConfirm);
            const invalidLength = (passwordNew && passwordNew.length < config['core.register.minPasswordLength']);

            if (passwordSet) {
                if (passwordNotEquals) {
                    this.createNotificationError({
                        title: this.$tc('sw-customer.detail.titleSaveError'),
                        message: this.$tc('sw-customer.detail.notificationPasswordErrorMessage')
                    });

                    return false;
                } else if (invalidLength) {
                    this.createNotificationError({
                        title: this.$tc('sw-customer.detail.titleSaveError'),
                        message: this.$tc('sw-customer.detail.notificationPasswordLengthErrorMessage')
                    });

                    return false;
                }
            }

            return true;
        },
```

That means I've made onSave async as well. I'm mainly a PHP guy, so if there's a way to do this better, please let me know :-)

Reply:
@wexotk Did you push those changes?

Reply:
@raknison no not yet, figured you might have some input, but I can push them now.

Reply:
@wexotk Nice work! - I am happy with this refactoring.